File: tests/fixtures/lua-simple/src/utils/logger.lua
-- Simple logging utility.
local Logger = {}
Logger.__index = Logger

function Logger.new(context)
    local self = setmetatable({}, Logger)
    self.context = context
    return self
end

function Logger:info(message)
    print("[INFO] " .. self.context .. ": " .. message)
end

function Logger:error(message)
    io.stderr:write("[ERROR] " .. self.context .. ": " .. message .. "\n")
end

function Logger:debug(message)
    print("[DEBUG] " .. self.context .. ": " .. message)
end

return Logger

File: tests/fixtures/php-simple/src/Utils/Logger.php
<?php

namespace App\Utils;

/** Simple logging utility. */
class Logger
{
    private string $context;

    public function __construct(string $context)
    {
        $this->context = $context;
    }

    public function info(string $message): void
    {
        echo "[INFO] {$this->context}: {$message}\n";
    }

    public function error(string $message): void
    {
        fwrite(STDERR, "[ERROR] {$this->context}: {$message}\n");
    }

    public function debug(string $message): void
    {
        echo "[DEBUG] {$this->context}: {$message}\n";
    }
}
